objcgen/processor.cs

```diff
@@ -67,7 +67,7 @@ public void Process (ProcessedAssembly a)
 
 		protected void AddExtraType (ProcessedType pt)
 		{
-			typeQueue.Enqueue (pt);
+			Types.Add (pt);
```

Comment:
Do you mean create a queue in the AddExtraType method here and add types to it (and clear it)?  That would cause issues when it later (if called while processing an assembly) calls ```typeQueue = new Queue<ProcessedType> (Types);```.

I don't see an easy way to add the contents of Types to an existing queue without having to iterate over Types and add them one at a time.

What about a null check for typeQueue and if it's null add to Types (should catch when processing assembly before the type is created), otherwise add it to the queue?